internal/controllers/import_controller.go

```diff
@@ -257,11 +269,17 @@ func (r *CAPIImportReconciler) shouldAutoImport(ctx context.Context, capiCluster
 	return autoImport, nil
 }
 
-func (r *CAPIImportReconciler) reconcileDelete(ctx context.Context, _ *clusterv1.Cluster) (ctrl.Result, error) {
+func (r *CAPIImportReconciler) reconcileDelete(ctx context.Context, capiCluster *clusterv1.Cluster) (ctrl.Result, error) {
 	log := log.FromContext(ctx)
 	log.Info("Reconciling CAPI Cluster Delete")
 
-	// TODO: add implementation
+	annotations := capiCluster.GetAnnotations()
+	if annotations == nil {
+		annotations = map[string]string{}
+	}
+
+	annotations[clusterImportedAnnotation] = "true"
```

Comment:
Should we annotate the rancher cluster instead?

Reply:
based on the discussions, we agreed to annotate capi cluster, no?

>2. Set the annotation on Rancher cluster deletion to the CAPI cluster manifest.

Reply:
Yes we agreed to annotate the CAPI cluster as the Rancher Cluster will have been deleted.

Reply:
`reconcileDelete` will only be called if the CAPI cluster has a non-zero deletion timestamp. So what will happen here is that we will annotate the capi cluster and then it will be deleted.

As we are also watching Rancher Clusters we could change object where we are [checking the deletion timestamp ](https://github.com/rancher-sandbox/rancher-turtles/blob/2ccb252757b2febd270ae1bf3b7383565c45a3a3/internal/controllers/import_controller.go#L154) to be the Rancher Cluster instead and then call **reconcileDelete** passing in the CAPI cluster & Rancher Cluster. If the Rancher Cluster has our annotation that said we imported it, then we can then annotate the CAPI cluster to say don't auto-import again.

Reply:
@richardcase thanks for review, I changed `reconcileDelete` logic based on your suggestions, PTAL